xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -592,11 +592,21 @@ bool CWinSystemWayland::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, boo
       ApplyBufferScale(m_scale);
     }
 
-    // FIXME This assumes that the resolution has already been set. Should
-    // be moved to some post-change callback when resolution setting is refactored.
-    if (!m_inhibitSkinReload)
+    if (m_shellSurfaceState.test(IShellSurface::STATE_RESIZING) && !m_resizeSkinReloadTimer.IsRunning())
     {
-      g_application.ReloadSkin();
+      // Reloading the skin is costly -> during resizing (when the size changes quite
```

Comment:
You need to reload for the new aspect ratio to apply. It's broken for me on X11: https://ibb.co/mTs5Uv